endless-key-builder

```diff
@@ -178,9 +200,15 @@ class EKB(object):
         config_files = [
             # Host settings
             os.path.join(self.sysconfdir, 'private.ini'),
-            # User settings
-            os.path.join(self.userconfdir, 'private.ini'),
         ]
+        if self.local_configdir:
+            config_files.extend([
+                os.path.join(self.local_configdir, 'private.ini'),
+            ])
+        # User settings
+        config_files.extend([
+            os.path.join(self.userconfdir, 'private.ini'),
+        ])
```

Comment:
This block might not be needed. It's not loading `os.path.join(self.configdir, 'private.ini')`, so it would be consistent to not do that for `os.path.join(self.local_configdir, 'private.ini')`. I.e., the only private settings come from the system or user configuration.

However, if you do want to keep that, I think it doesn't quite match the order for the non-private config files. I think if the precedence is going to match the above order, `self.local_configdir` should go before both `self.sysconfdir` and `self.userconfdir`. So, I think this part should be a bit more like:

```
config_files = [os.path.join(self.configdir, 'private.ini')]
if self.local_configdir:
    config_files.append(os.path.join(self.local_configdir, 'private.ini'))
config_file.extend([
    # Host settings
    os.path.join(self.sysconfdir, 'private.ini'),
    # User settings
    os.path.join(self.userconfdir, 'private.ini'),
])

Reply:
Good catch! Let's keep it and make it consistent. The only change from your suggestion above was to not include `os.path.join(self.configdir, 'private.ini')` since it does not make sense to have private.ini one a public repo (but someone may want it on a private repo).